segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/ZooKeeperServiceRunner.java

```diff
@@ -106,19 +163,134 @@ public void stop() {
+     *                             Empty string if `secureZk` is false or a password does not exist.
+     * @param trustStorePasswordPath Location of password path for trust store.
+     *                               Empty string if `secureZk` is false or a password does not exist.
      * @return True if ZooKeeper started within a specified timeout, false otherwise.
      */
+    public static boolean waitForServerUp(int zkPort, boolean secureZk, String trustStore, String keyStore,
+                                          String keyStorePasswordPath, String trustStorePasswordPath) {
+        val address = "localhost:" + zkPort;
+        if (secureZk) {
+            return waitForSSLServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT, trustStore, keyStore,
+                    keyStorePasswordPath, trustStorePasswordPath);
+        } else {
+            return LocalBookKeeper.waitForServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT);
+        }
+    }
+
     public static boolean waitForServerUp(int zkPort) {
-        val address = LOOPBACK_ADDRESS.getHostAddress() + ":" + zkPort;
-        return LocalBookKeeper.waitForServerUp(address, LocalBookKeeper.CONNECTION_TIMEOUT);
+        return waitForServerUp(zkPort, false, "", "", "", "");
+    }
+
+    private static boolean waitForSSLServerUp(String address, long timeout, String trustStore, String keyStore,
+                                              String keyStorePasswdPath, String trustStorePasswordPath) {
+        long start = MathUtils.now();
+        String[] split = address.split(":");
+        String host = split[0];
+        int port = Integer.parseInt(split[1]);
+
+        while (true) {
+            try {
+                SSLContext context = SSLContext.getInstance("TLS");
+                TrustManagerFactory trustManager = getTrustManager(trustStore, trustStorePasswordPath);
+                KeyManagerFactory keyFactory = getKeyManager(keyStore, keyStorePasswdPath);
+
+                context.init(keyFactory.getKeyManagers(), trustManager.getTrustManagers(), null);
+                SSLSocketFactory factory = context.getSocketFactory();
+
+                Socket sock = context.getSocketFactory().createSocket(new Socket(host, port), host, port, true);
+                BufferedReader reader = null;
+                try {
+                    OutputStream outstream = sock.getOutputStream();
+                    outstream.write("stat".getBytes());
+                    outstream.flush();
+
+                    reader =
+                            new BufferedReader(
+                                    new InputStreamReader(sock.getInputStream()));
+                    String line = reader.readLine();
+                    if (line != null && line.startsWith("Zookeeper version:")) {
+                        log.info("Server UP");
+                        return true;
+                    }
+                } catch (Exception e) {
+                  e.printStackTrace();
```

Comment:
Please get out of the habit of doing this. Log the exception. Never call `printStackTrace`